app/AssetManifest.scala

```diff
@@ -0,0 +1,75 @@
+package lila.app
+
+import play.api.{ Environment, Mode }
+import play.api.libs.json.{ JsObject, Json, JsValue, JsString }
+import java.nio.file.{ Files, Path, Paths }
+import java.time.Instant
+
+case class SplitAsset(name: String, imports: List[String])
+case class AssetMaps(js: Map[String, SplitAsset], css: Map[String, String])
+
+final class AssetManifest(environment: Environment):
+
+  private val filename = s"manifest.${if environment.mode == Mode.Prod then "prod" else "dev"}.json"
+  private val pathname = environment.getFile(s"conf/$filename").toPath
+  private var lastModified: Instant = Instant.MIN
+  private var maps: AssetMaps       = AssetMaps(Map.empty, Map.empty)
+  private val keyRe                 = """^(?!common\.)(\S+)\.([A-Z0-9]{8})\.(?:js|css)""".r
+
+  def js(key: String): Option[SplitAsset]    = maps.js.get(key)
+  def css(key: String): Option[String]       = maps.css.get(key)
+  def deps(keys: List[String]): List[String] = keys.flatMap { key => js(key).so(_.imports) }.distinct
+
+  def update: AssetManifest =
+    val current = Files.getLastModifiedTime(pathname).toInstant
```

Comment:
are we checking all these files modified times on every single page load?

Reply:
No it's a stat of the asset manifest file itself. It's not necessary on prod though, I assume we'll [call AssetManifest reload here](https://github.com/lichess-org/lila/blob/e6acdd905119dc6c6d160f72316a71b919f018c1/modules/api/src/main/Cli.scala#L33)

I just wasn't sure it's worthwhile to be optimizing yet given that I assume half of this will probably be rewritten if it's merged. If getLastModifiedTime every request on dev deploys bothers you, I could go with WatchService which (on linux) would hopefully use inotify rather than consult the inode cache.